core/src/main/java/bisq/core/offer/AtomicOfferPayload.java

```diff
@@ -0,0 +1,207 @@
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Instance fields
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private final String id;
+    private final long date;
+    private final NodeAddress ownerNodeAddress;
+    @JsonExclude
+    private final PubKeyRing pubKeyRing;
+    private final Direction direction;
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Constructor
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    public AtomicOfferPayload(String id,
+                              long date,
+                              NodeAddress ownerNodeAddress,
+                              PubKeyRing pubKeyRing,
+                              Direction direction) {
+        this.id = id;
+        this.date = date;
+        this.ownerNodeAddress = ownerNodeAddress;
+        this.pubKeyRing = pubKeyRing;
+        this.direction = direction;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // PROTO BUFFER
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    public static protobuf.OfferDirection toProtoMessage(Direction direction) {
+        return protobuf.OfferDirection.valueOf(direction.name());
+    }
+    public static Direction fromProto(protobuf.OfferDirection offerDirection) {
+        return ProtoUtil.enumFromProto(Direction.class, offerDirection.name());
+    }
+
+    @Override
+    public protobuf.StoragePayload toProtoMessage() {
+        protobuf.AtomicOfferPayload.Builder builder = protobuf.AtomicOfferPayload.newBuilder()
+                .setId(id)
+                .setDate(date)
+                .setOwnerNodeAddress(ownerNodeAddress.toProtoMessage())
+                .setPubKeyRing(pubKeyRing.toProtoMessage())
+                .setDirection(toProtoMessage(direction));
+
+        return protobuf.StoragePayload.newBuilder().setAtomicOfferPayload(builder).build();
+    }
+
+    public static AtomicOfferPayload fromProto(protobuf.AtomicOfferPayload proto) {
+        return new AtomicOfferPayload(proto.getId(),
+                proto.getDate(),
+                NodeAddress.fromProto(proto.getOwnerNodeAddress()),
+                PubKeyRing.fromProto(proto.getPubKeyRing()),
+                fromProto(proto.getDirection()));
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // API
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public long getAmount() {
+        return 0;
+    }
+
+    @Override
+    public long getMinAmount() {
+        return 0;
+    }
+
+    @Override
+    public String getBaseCurrencyCode() {
+        return "BSQ";
```

Comment:
as this is an BSQ atomic offer payload shall we call it AtomicBsqOfferPayload in case we want to support more Atomic offers in the future (XMR?)